Bottom: 200a9d6f3734840cccc9509cdd05797325c3f2a9
Top:    8ca16a758a17329d2c04b031f337c3d806ff28a3
Author: Lawrence D'Anna <lawrence_danna@apple.com>
Date:   2019-09-19 13:02:10 -0700

convert SBDebugger::***FileHandle() wrappers to native files.


---
diff --git a/lldb/scripts/interface/SBDebugger.i b/lldb/scripts/interface/SBDebugger.i
index 3851c428bfb..484e6eb16b4 100644
--- a/lldb/scripts/interface/SBDebugger.i
+++ b/lldb/scripts/interface/SBDebugger.i
@@ -165,29 +165,41 @@ public:
     void
     SkipLLDBInitFiles (bool b);
 
-    %feature("autodoc", "DEPRECATED, use SetInputFile");
-    void
-    SetInputFileHandle (FILE *f, bool transfer_ownership);
-
-    %feature("autodoc", "DEPRECATED, use SetOutputFile");
-    void
-    SetOutputFileHandle (FILE *f, bool transfer_ownership);
-
-    %feature("autodoc", "DEPRECATED, use SetErrorFile");
-    void
-    SetErrorFileHandle (FILE *f, bool transfer_ownership);
-
-    %feature("autodoc", "DEPRECATED, use GetInputFile");
-    FILE *
-    GetInputFileHandle ();
-
-    %feature("autodoc", "DEPRECATED, use GetOutputFile");
-    FILE *
-    GetOutputFileHandle ();
-
-    %feature("autodoc", "DEPRECATED, use GetErrorFile");
-    FILE *
-    GetErrorFileHandle ();
+    %extend {
+
+        %feature("autodoc", "DEPRECATED, use SetInputFile");
+        void
+        SetInputFileHandle (lldb::FileSP file,
+                            bool transfer_ownership) {
+            self->SetInputFile(file);
+        }
+
+        %feature("autodoc", "DEPRECATED, use SetOutputFile");
+        void
+        SetOutputFileHandle (lldb::FileSP file,
+                             bool transfer_ownership) {
+            self->SetOutputFile(file);
+        }
+
+        %feature("autodoc", "DEPRECATED, use SetErrorFile");
+        void
+        SetErrorFileHandle (lldb::FileSP file,
+                             bool transfer_ownership) {
+            self->SetErrorFile(file);
+        }
+
+        lldb::FileSP GetInputFileHandle() {
+            return self->GetInputFile().GetFile();
+        }
+
+        lldb::FileSP GetOutputFileHandle() {
+            return self->GetOutputFile().GetFile();
+        }
+
+        lldb::FileSP GetErrorFileHandle() {
+            return self->GetErrorFile().GetFile();
+        }
+    }
 
     SBError
     SetInputFile (SBFile file);
